grakn-core/src/main/java/ai/grakn/util/Schema.java

```diff
@@ -145,11 +151,11 @@ public Class getClassType(){
      */
     public enum ConceptProperty {
         //Unique Properties
-        TYPE_LABEL(String.class), INDEX(String.class), ID(String.class),
+        TYPE_LABEL(String.class), INDEX(String.class), ID(String.class), TYPE_ID(Integer.class),
 
         //Other Properties
-        TYPE(String.class), IS_ABSTRACT(Boolean.class), IS_IMPLICIT(Boolean.class), IS_SHARD(Boolean.class),
-        REGEX(String.class), DATA_TYPE(String.class), INSTANCE_COUNT(Long.class),
+        INSTANCE_TYPE_ID(Integer.class), IS_ABSTRACT(Boolean.class), IS_IMPLICIT(Boolean.class), IS_SHARD(Boolean.class),
+        REGEX(String.class), DATA_TYPE(String.class), INSTANCE_COUNT(Long.class), CURRENT_TYPE_ID(Integer.class),
```

Comment:
Ahhhh my shame :( . That stores the current number of types we have in the graph and it is persisted ONLY on the meta type. Thats how we know what to use as the next id.